connection/connection.go

```diff
@@ -105,29 +106,59 @@ func ExitOnConnectionLoss() func() bool {
 	}
 }
 
+// withTimeout adds a configurable timeout on the gRPC calls.
+func withTimeout(timeout time.Duration) Option {
+	return func(o *options) {
+		o.timeout = timeout
+	}
+}
+
+// withMetrics enables the recording of metrics on the gRPC calls with the provided CSIMetricsManager.
+func withMetrics(metricsManager metrics.CSIMetricsManager) Option {
+	return func(o *options) {
+		o.metricsManager = metricsManager
+	}
+}
+
+// WithOtelTracing enables the recording of traces on the gRPC calls with opentelemetry gRPC interceptor.
+func WithOtelTracing() Option {
+	return func(o *options) {
+		o.enableOtelTracing = true
+	}
+}
```

Comment:
btw why are `withTimeout` and `withMetrics` unexported but `WIthOtelxxx` is exported?

Reply:
Mainly because they were already set when calling `Connect` with the previous implementation and I did not want to let the user override it (because there was no need until now) and if the need were to arise, the functions could simply be exported.
But I can also directly export them and let them override the default I set in `Connect` if you prefer.

Reply:
I exported the function and let them override the default set in `Connect`